File: todo_mobile/lib/models/practice_generation_task.dart
class PracticeGenerationTask {
  final int id;
  final String status; // "PENDING", "COMPLETED", "FAILED"
  final String summary;
  final String? questionsJson;
  final String? errorMessage;
  final String createdAt;

  PracticeGenerationTask({
    required this.id,
    required this.status,
    required this.summary,
    this.questionsJson,
    this.errorMessage,
    required this.createdAt,
  });

  factory PracticeGenerationTask.fromJson(Map<String, dynamic> json) {
    return PracticeGenerationTask(
      id: json['id'],
      status: json['status'],
      summary: json['summary'],
      questionsJson: json['questionsJson'], // 后端返回的是 JSON 字符串
      errorMessage: json['errorMessage'],
      createdAt: json['createdAt'],
    );
  }
}

File: todo_service/src/main/java/com/suat/app/backend/todo_service/entity/PracticeGenerationTask.java
package com.suat.app.backend.todo_service.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.Instant;

@Entity
@Table(name = "practice_generation_task")
@Getter
@Setter
public class PracticeGenerationTask {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "app_user_id", nullable = false)
    private AppUser appUser;

    @Column(nullable = false)
    private String status; // "PENDING", "COMPLETED", "FAILED"

    @Column(nullable = false)
    private String summary; // e.g. "Easy / Java / QA"

    @Column(name = "questions_json", columnDefinition = "TEXT")
    private String questionsJson;

    @Column(name = "error_message", columnDefinition = "TEXT")
    private String errorMessage;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = Instant.now();
        updatedAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now();
    }
}